src/util/modals.js

```diff
@@ -40,3 +40,17 @@ export const hideModal = () => {
 
 export const modalCancelButton = dom('button', null, { click: hideModal }, ['Cancel']);
 export const modalCompleteButton = dom('button', { class: 'blue' }, { click: hideModal }, ['OK']);
+
+export const showErrorModal = exception => {
+  console.error('XKit Rewritten error:', exception);
+
+  showModal({
+    title: 'Something went wrong.',
+    message: [
+      [exception.body?.errors?.[0]?.detail, exception.errors?.[0]?.detail, exception.message]
+        .filter(Boolean)
+        .join('\n\n')
```

Comment:
Hmm, do we want to show more than one message...? Logging the exception to the console feels "good enough" for debugging purposes...

```suggestion
        .find(Boolean)
```

Reply:
Hm, that sounds reasonable enough. Just a matter of picking the order correctly so that the useful one is surfaced, then.

I am personally strongly against only giving the user the "friendly" message, because it tells both them and the developer they report the issue to absolutely nothing, and the amount of time I've spent trying to fix XKit 7's Editable Reblogs that was just spent not getting any information about which error it was... okay fine yes I didn't need to do that and so I can't really use that as an argument, but.

![Screenshot 2024-02-20 at 6 25 14 PM](https://github.com/AprilSylph/XKit-Rewritten/assets/8336245/18be85d0-a82f-4f70-b4e2-6a23e42314bd)

Reply:
Hmm, right, the friendly messages are... very bare, huh. Nevermind, this is fine as-is.